perl-package/AI-MXNetCAPI/mxnet_typemaps.i

```diff
@@ -300,6 +302,37 @@
     Safefree($1);
 }
 
+%typemap(in) (dim_t *in)
+{
+    AV *tempav;
+    int i;
+    SV  **tv;
+    int av_len; 
+    if (!SvROK($input))
+        croak("Argument $argnum is not a reference.");
+        if (SvTYPE(SvRV($input)) != SVt_PVAV)
+        croak("Argument $argnum is not an array.");
+        tempav = (AV*)SvRV($input);
+    av_len = av_len(tempav) + 1;
+    if(av_len)
+    {
+        $1 = (dim_t *)safemalloc(av_len*sizeof(dim_t));
```

Comment:
This use of `safemalloc()` may be better as `Newxc($1, av_len, dim_t *)`.  `safemalloc()` is not documented in perlapi, I think it just happens to be the default in most perl compilations.

Reply:
Ok, thanks for this observation, I'll cleanup safemalloc usage up in future, need to verify it's present in all perls from 5.14 till 5.28

Reply:
Yeah, it's been around, at least as far back as [v5.8.8](https://perldoc.perl.org/5.8.8/perlapi.html#Newxc).